Problem copying file when subfolder doesn't exist

I had this problem on Android 5.1.1 (a Kindle) when doing a fresh install. My best guess is because the folder "databases" does not exist so it couldn't open the new file to copy it.

However, that's just a guess...

```
    java.lang.RuntimeException: Unable to create application com.egghead.LogicApp: java.lang.RuntimeException: java.io.FileNotFoundException: /data/data/com.eggheadgames.logicproblems/databases/data.db: open failed: ENOENT (No such file or directory)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4883)
        at android.app.ActivityThread.access$1500(ActivityThread.java:160)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1382)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:135)
        at android.app.ActivityThread.main(ActivityThread.java:5597)
        at java.lang.reflect.Method.invoke(Native Method)
        at java.lang.reflect.Method.invoke(Method.java:372)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
     Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: /data/data/com.eggheadgames.logicproblems/databases/data.db: open failed: ENOENT (No such file or directory)
        at com.eggheadgames.assethelper.OsUtil.loadFileToLocalStorage(OsUtil.java:42)
        at com.eggheadgames.assethelper.AssetHelper.loadFileToStorage(AssetHelper.java:119)
        at com.eggheadgames.assethelper.AssetHelper.copyIfNew(AssetHelper.java:42)
        at com.egghead.provider.ContentProvider.init(ContentProvider.java:45)
        at com.egghead.LogicApp.onCreate(LogicApp.java:47)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1015)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4880)
        at android.app.ActivityThread.access$1500(ActivityThread.java:160) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1382) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5597) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
     Caused by: java.io.FileNotFoundException: /data/data/com.eggheadgames.logicproblems/databases/data.db: open failed: ENOENT (No such file or directory)
        at libcore.io.IoBridge.open(IoBridge.java:456)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:89)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:74)
        at com.eggheadgames.assethelper.OsUtil.loadFileToLocalStorage(OsUtil.java:32)
        at com.eggheadgames.assethelper.AssetHelper.loadFileToStorage(AssetHelper.java:119) 
        at com.eggheadgames.assethelper.AssetHelper.copyIfNew(AssetHelper.java:42) 
        at com.egghead.provider.ContentProvider.init(ContentProvider.java:45) 
        at com.egghead.LogicApp.onCreate(LogicApp.java:47) 
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1015) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4880) 
        at android.app.ActivityThread.access$1500(ActivityThread.java:160) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1382) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5597) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
     Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
        at libcore.io.Posix.open(Native Method)
        at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
        at libcore.io.IoBridge.open(IoBridge.java:442)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:89) 
        at java.io.FileOutputStream.<init>(FileOutputStream.java:74) 
        at com.eggheadgames.assethelper.OsUtil.loadFileToLocalStorage(OsUtil.java:32) 
        at com.eggheadgames.assethelper.AssetHelper.loadFileToStorage(AssetHelper.java:119) 
        at com.eggheadgames.assethelper.AssetHelper.copyIfNew(AssetHelper.java:42) 
        at com.egghead.provider.ContentProvider.init(ContentProvider.java:45) 
        at com.egghead.LogicApp.onCreate(LogicApp.java:47) 
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1015) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4880) 
        at android.app.ActivityThread.access$1500(ActivityThread.java:160) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1382) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:135) 
        at android.app.ActivityThread.main(ActivityThread.java:5597) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at java.lang.reflect.Method.invoke(Method.java:372) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779) 
```

--- a/build.gradle
+++ b/build.gradle
@@ -1,8 +1,8 @@
 buildscript {
     ext {
         app = [
-                versionCode: 1,
-                versionName: "1.0.0"
+                versionCode: 2,
+                versionName: "1.0.1"
         ]
 
         general = [

--- a/library/src/main/java/com/eggheadgames/assethelper/OsUtil.java
+++ b/library/src/main/java/com/eggheadgames/assethelper/OsUtil.java
@@ -46,6 +46,14 @@ public class OsUtil {
     }
 
     public String generateFilePath(String destinationFolder, String fileName) {
+        File folder = new File(destinationFolder);
+        if (!folder.exists()) {
+            boolean created = folder.mkdirs();
+            if (!created) {
+                throw new RuntimeException("Can not create folder. Path: " + destinationFolder);
+            }
+        }
+
         return destinationFolder + File.separator + fileName;
     }
 

